src/components/Todo.jsx

```diff
@@ -0,0 +1,260 @@
+import { useState } from 'react'
+
+const Todo = () => {
+
+    // states
+    const [tasksState, setTaskState] = useState([])
+    const [taskTitleState, setTaskTitleState] = useState("")
+    const [todoListName, setTodoListName] = useState("My Todo List")
+    const [todoListNameHolder, setTodoListNameHolder] = useState("")
+    const [todoListTasksAmount, setTodoListTasksAmount] = useState("")
+    const [isFiltered, setIsFiltered] = useState(false)
+    const [preFilteredList, setPreFilteredList] = useState([])
+
+    // new task template prototype
+    function Task(title) {
+        this.title = title;
+        this.completed = false;
+        this.createdAt = Date.now()
+        this.completedAt = null;
+        this.isPriority = false;
+        this.id = Math.round((Math.floor(Math.random() * 10000000000) + 1) / 1) * 1
```

Comment:
Although I know you're created a random ID here, this could be pulled out into a function to be a bit more clear of its intent. Also, while this isn't a production app, its not guarunteed you won't create clashing IDs, which would be a huge bug. You could write a smarter ID generator fn to avoid this, or more commonly just reach more a library to do such things for you